lib/posix/mqueue.c

```diff
@@ -251,7 +250,10 @@ int mq_timedsend(mqd_t mqdes, const char *msg_ptr, size_t msg_len,
 	mqueue_desc *mqd = (mqueue_desc *)mqdes;
 	s32_t  timeout;
 
-	timeout = (s32_t) timespec_to_timeoutms(abstime);
+	timeout = (s32_t) _timespec_to_timeoutms(abstime);
+	if (timeout <= 0)
+		return ETIMEDOUT;
```

Comment:
Well, you'd need to follow the codestyle: braces are mandatory in all cases, even for a single statement. (Here and below.)